lab-anthony/test/note-route-test.js

```diff
@@ -0,0 +1,147 @@
+      });
+      return;
+    }
+    done();
+  });
+
+  describe('testing method POST on endpoint /api/note', function(){
+    describe('testing successful POST request', function(){
+      before((done)=> {
+        request.post(`${serverUrl}/api/note`)
+          .send({content: 'This is a test note!'})
+          .end((err,res)=>{
+            this.res = res;
+            this.note = res.body;
+            done();
+          });
+      });
+
+      it('should return status 200', ()=>{
+        expect(this.res.status).to.equal(200);
+      });
+
+      it('should return a note', ()=>{
+        expect(this.note.content).to.equal('This is a test note!');
+      });
+    });
+  });
+
+  describe('testing method POST', function() {
+    describe('testing bad POST request', function(){
+      before((done)=> {
+        request.post(`${serverUrl}/api/note`)
+          .send({})
+          .end((err,res)=>{
+            this.res = res;
+            this.note = res.body;
+            done();
+          });
+      });
+      it('should return status 400', ()=>{
+        expect(this.res.status).to.equal(400);
+      });
+    });
+  });
+
+  describe('testing method GET on endpoint /api/note', function(){
+    describe('testing successful GET request', function() {
+      before((done)=> {
+        request.post(`${serverUrl}/api/note`)
+          .send({content: 'This is a test note!'})
+          .end((err,res)=>{
+            request.get(`${serverUrl}/api/note?id=${res.body.id}`)
+            .end((err, res)=>{
+              this.res = res;
+              done();
+            });
+          });
+      });
+      it('should return status 200', ()=>{
+        expect(this.res.status).to.equal(200);
+      });
+    });
+  });
+
+  describe('testing method GET on endpoint /api/note', function(){
+    describe('testing bad GET request', function() {
+      before((done)=> {
+        request.post(`${serverUrl}/api/note`)
+          .send({content: 'This is a test note!'})
+          .end(()=>{
+            request.get(`${serverUrl}/api/note`)
+            .end((err, res)=>{
+              this.res = res;
+              done();
+            });
+          });
+      });
+      it('should return status 400', ()=>{
+        expect(this.res.status).to.equal(400);
+      });
+    });
+  });
+
+  describe('testing method GET on endpoint /api/note', function(){
+    describe('testing bad GET request', function() {
+      before((done)=> {
+        request.post(`${serverUrl}/api/note`)
+          .send({content: 'This is a test note!'})
+          .end((err, res)=>{
+            request.get(`${serverUrl}/api/note?id=${res.body.id + 1}`)
+            .end((err, res)=>{
+              this.res = res;
+              done();
+            });
+          });
+      });
+      it('should return status 404', ()=>{
+        expect(this.res.status).to.equal(404);
+      });
+    });
+  });
+
+  describe('testing method GET on endpoint /api/note/all', function(){
```

Comment:
This doesn't test that your server gives you back anything meaningful. Even if you don't save all of the ids from all of your tests, you could at least check that the array you're getting back has something in it, or that it has the item that you just posted in it.